lib/ssl/src/dtls_socket.erl

```diff
@@ -51,12 +51,16 @@ listen(Port, #config{inet_ssl = SockOpts,
                      ssl = SslOpts,
                      emulated = EmOpts,
                      inet_user = Options} = Config) ->
-    IP = proplists:get_value(ip, SockOpts, {0,0,0,0}),
+    DefIP = case proplists:get_value(inet6, SockOpts, false) of
+                false -> {0,0,0,0};
+                true -> {0,0,0,0, 0,0,0,0}
+            end,
```

Comment:
consider having `default_ip(SockOpts)` helper function, so DefIP variable is maybe not needed and we don't repeat code.